benchmarkcommon/src/test/java/com/ss/benchmark/httpclient/HCPerformanceTests.java

```diff
@@ -0,0 +1,395 @@
+
+        logger.info("Completed " + method);
+    }
+
+    @Test(groups = {"sync", "blocking"})
+    public void testSyncBlockingLongLongPOST() {
+        String method = myName();
+        logger.info("Start " + method);
+
+        for (int i = 0; i < EXECUTIONS; i++) {
+            syncPOST(ECHO_DELAY_LONG_URL,
+                    Payloads.LONG_JSON,
+                    Payloads.LONG_JSON,
+                    metricRegistry.timer(MetricRegistry.name(this.getClass(), method, "timing")),
+                    metricRegistry.counter(MetricRegistry.name(this.getClass(), method, "errorRate")));
+        }
+        logger.info("Completed " + method);
+    }
+
+    @Test(invocationCount = EXECUTIONS, threadPoolSize = 100, groups = {"async", "blocking"})
+    public void testAsyncBlockingLongLongPOST() {
+        String method = myName();
+        logger.info("Start " + method);
+
+        syncPOST(ECHO_DELAY_LONG_URL,
+                Payloads.LONG_JSON,
+                Payloads.LONG_JSON,
+                metricRegistry.timer(MetricRegistry.name(this.getClass(), method, "timing")),
+                metricRegistry.counter(MetricRegistry.name(this.getClass(), method, "errorRate")));
+
+        logger.info("Completed " + method);
+    }
+
+    @Test(groups = {"sync", "nonblocking"})
+    public void testSyncNonBlockingShortGET() {
+        String method = myName();
+        logger.info("Start " + method);
+
+        CountDownLatch latch = new CountDownLatch(EXECUTIONS);
+
+        for (int i = 0; i < EXECUTIONS; i++) {
+            asyncGET(latch,
+                    metricRegistry.timer(MetricRegistry.name(this.getClass(), method, "timing")),
+                    metricRegistry.counter(MetricRegistry.name(this.getClass(), method, "errorRate")));
+        }
+
+        logger.info("Completed " + method);
+    }
+
+    @Test(invocationCount = EXECUTIONS, threadPoolSize = 100, groups = {"async", "nonblocking"})
+    public void testAsyncNonBlockingShortGET() {
+        String method = myName();
+        logger.info("Start " + method);
+
+        asyncGET(new CountDownLatch(1),
+                metricRegistry.timer(MetricRegistry.name(this.getClass(), method, "timing")),
+                metricRegistry.counter(MetricRegistry.name(this.getClass(), method, "errorRate")));
+
+        logger.info("Completed " + method);
+    }
+
+    @Test(groups = {"sync", "nonblocking"})
+    public void testSyncNonBlockingShortShortPOST() {
+        String method = myName();
+        logger.info("Start " + method);
+
+        CountDownLatch latch = new CountDownLatch(EXECUTIONS);
+
+        for (int i = 0; i < EXECUTIONS; i++) {
+            asyncPOST(ECHO_DELAY_SHORT_URL,
+                    Payloads.SHORT_JSON,
+                    Payloads.SHORT_JSON,
+                    latch,
+                    metricRegistry.timer(MetricRegistry.name(this.getClass(), method, "timing")),
```

Comment:
We repeat this quite a bit.  We could repeat something like this:
```
timingTimer(method)
errorRateCounter(method)
```

Reply:
let's leave for now